server/api/personalGoal/personalGoal.controller.js

```diff
@@ -37,14 +37,18 @@ const updatePersonalGoal = async (req, res) => {
       set.dateCompleted = Date.now()
       break
   }
-  await PersonalGoal.updateOne({ _id: req.body._id }, { $set: set }).exec()
+  try {
+    await PersonalGoal
+      // .accessibleBy(userAbility, Action.UPDATE)
+      .updateOne({ _id: id }, { $set: set }).exec()
```

Comment:
As mentioned above, I think this section is currently open to authenticated users updating any other users personal goals

Reply:
Resolved by putting the accessibleBy back in.  Using accessibleBy does not cause a 403 Forbidden to be returned. Instead the call returns 200 OK but does not make the changes to the database.  This is ok and the tests can reflect it - but there is a potential for confusing bugs.

Reply:
Yeah I've hit that problem elsewhere - I think I handled it something like this:

```
const personalGoal = await PersonalGoal
  .accessibleBy(req.ability, Action.UPDATE)
  .findOne({ _id: id })

if (personalGoal === null) {
  // 403 or 404 depending on preference
}

// update query here
```

The downside being you're doing 2 queries.

Alternatively, it looks like `.updateOne` returns a result object that could be used to check if records were updated by the query

[Mongoose Model.updateOne API doc](https://mongoosejs.com/docs/api/model.html#model_Model.updateOne)